Sources/MapNode.swift

```diff
@@ -8,28 +8,33 @@
 import Stencil
 
 class MapNode: NodeType {
-  let variable: Variable
+  let resolvable: Resolvable
   let resultName: String
   let mapVariable: String?
   let nodes: [NodeType]
+  let token: Token?
 
   class func parse(parser: TokenParser, token: Token) throws -> NodeType {
     let components = token.components()
 
-    guard components.count == 4 && components[2] == "into" ||
-      components.count == 6 && components[2] == "into" && components[4] == "using" else {
+    func hasToken(_ token: String, at index: Int) -> Bool {
+      return components.count > (index + 1) && components[index] == token
```

Comment:
I nitpick but I feel it's more readable to write it as `index < components.count - 1` instead of the other way around

Reply:
Yeah I just copied this from stencil (seemed clearer instead of checking a whole bunch of indices). Will update this.

Reply:
Another way is to write `collection.indices ~= index` (or is it the other way around I always forget)

Reply:
That's the first time I see that (the match operator) used for checking bounds, interesting.

Reply:
Well I haven't checked if that works on indices (which is what? A collection?), I know it works on Ranges but didn't check in code that it works work with indices… so curious if that's actually possible :wink: